File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/infrastructure/persistence/adapters/IssuedTokenAuditRepositoryAdapter.java
package br.com.deltaglobalbank.identity.infrastructure.persistence.adapters;

import java.util.UUID;

import br.com.deltaglobalbank.identity.domain.token.IssuedTokenAudit;
import br.com.deltaglobalbank.identity.domain.token.IssuedTokenAuditRepository;
import br.com.deltaglobalbank.identity.infrastructure.persistence.entities.IssuedTokenAuditEntity;
import br.com.deltaglobalbank.identity.infrastructure.persistence.mappers.IssuedTokenAuditMapper;
import br.com.deltaglobalbank.identity.infrastructure.persistence.repositories.JpaIssuedTokenAuditRepository;
import org.springframework.stereotype.Component;

@Component
public class IssuedTokenAuditRepositoryAdapter implements IssuedTokenAuditRepository {

    private final JpaIssuedTokenAuditRepository jpaIssuedTokenAuditRepository;

    public IssuedTokenAuditRepositoryAdapter(JpaIssuedTokenAuditRepository jpaIssuedTokenAuditRepository) {
        this.jpaIssuedTokenAuditRepository = jpaIssuedTokenAuditRepository;
    }

    @Override
    public void save(IssuedTokenAudit audit) {
        jpaIssuedTokenAuditRepository.save(IssuedTokenAuditMapper.toEntity(audit));
    }

    @Override
    public IssuedTokenAudit findByJti(String jti) {
        IssuedTokenAuditEntity entity = jpaIssuedTokenAuditRepository.findByJti(UUID.fromString(jti));
        return entity != null ? IssuedTokenAuditMapper.toDomain(entity) : null;
    }
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/infrastructure/persistence/adapters/IssuedTokenAuditRepositoryAdapter.kt
package br.com.deltaglobalbank.identity.infrastructure.persistence.adapters

import br.com.deltaglobalbank.identity.domain.token.IssuedTokenAudit
import br.com.deltaglobalbank.identity.domain.token.IssuedTokenAuditRepository
import br.com.deltaglobalbank.identity.infrastructure.persistence.mappers.toDomain
import br.com.deltaglobalbank.identity.infrastructure.persistence.mappers.toEntity
import br.com.deltaglobalbank.identity.infrastructure.persistence.repositories.JpaIssuedTokenAuditRepository
import org.springframework.stereotype.Component
import java.util.UUID

@Component
class IssuedTokenAuditRepositoryAdapter(
    private val jpaIssuedTokenAuditRepository: JpaIssuedTokenAuditRepository
) : IssuedTokenAuditRepository {

    override fun save(audit: IssuedTokenAudit) {
        jpaIssuedTokenAuditRepository.save(audit.toEntity())
    }

    override fun findByJti(jti: String): IssuedTokenAudit? {
       return jpaIssuedTokenAuditRepository.findByJti(UUID.fromString(jti))?.toDomain()

    }
}
